### ISSUE TITLE
Feature batch: six price-derived features, starting with the intraday-range gap

### ISSUE THREAD

**Issue by thingsofleon** [thingsofleon/twopercent#110]
Unsticks #78. The overnight loop has run zero experiments a night since 2026-07-31 because the curated queue and the auto-search grid are both fully recorded, and #78's own diagnosis is that more hyperparameters cannot help.

## Why features, and why these

61 recorded configs across two model families converged on the same result:

| strategy | best lift | best AUC | configs |
|---|---|---|---|
| baseline_gbm_v1 | 2.178 | 0.7315 | 27 |
| xgb_gbm_v1 | 2.168 | 0.7331 | 34 |

Different algorithms, exhaustive hyperparameter sweep, same ceiling. That is a **feature** ceiling.

The 11 existing features are price/volume/breadth derivatives, and there is a conspicuous gap in them: **the label is an intraday RANGE event — did the high reach +2% above the open — and no feature measures intraday range.** `vol_20d` is close-to-close dispersion, a different quantity: a stock can have low close-to-close vol and a wide daily range. `cnt_2pct_20d` counts threshold crossings but carries no magnitude, so a name averaging 1.9% open-to-high and one averaging 0.3% look identical if neither crossed.

That gap is the strongest prior here, and the first two features address it directly.

## The batch

All six derive from `daily_returns` alone. No new ingestion, no new provider, no new failure modes.

| feature | definition | why it might predict a +2% touch |
|---|---|---|
| `range_20d` | mean of `(high - low) / open` over the trailing 20 bars | The label is a range event and nothing measures range. A name whose typical day spans 5% touches +2% far more often than one spanning 1%. |
| `high_return_mean_20d` | mean of `high_return` over the trailing 20 bars | The label's own quantity, lagged. Carries the magnitude `cnt_2pct_20d` discards. |
| `gap_prior` | `(open - prev_close) / prev_close` on the signal day | A name that gapped today is in play tomorrow. Uses the SIGNAL day's open, known at its close. |
| `days_since_2pct` | bars since the last touch event, capped at 20 | Recency, which `cnt_2pct_20d` cannot express — five touches last week and five three weeks ago count the same today. |
| `volume_accel` | 5-bar mean volume / 20-bar mean volume | Distinguishes building interest from a one-day spike, which `volume_ratio` cannot. |
| `dist_52w_high` | `close / max(close)` over the trailing 252 bars | Position in the annual range; the standard breakout/momentum axis, absent today. |

## Timing (validate-new-strategy section 1)

Every one is computable from bars through the END of signal_date S, all trailing windows ending at S inclusive:

| feature | source | known at |
|---|---|---|
| `range_20d` | `daily_returns` high/low/open, bars <= S | S close |
| `high_return_mean_20d` | `daily_returns.high_return`, bars <= S | S close |
| `gap_prior` | S's open vs S-1's close | S open (so certainly by S close) |
| `days_since_2pct` | touch events on bars <= S | S close |
| `volume_accel` | volume, bars <= S | S close |
| `dist_52w_high` | close, bars <= S | S close |

No `LEAD`. Nothing reads the target day. `gap_prior` deliberately uses the SIGNAL day's open, never the target day's — the target open is unknown at the pre-open prediction moment and is the classic trap here.

## BLOCKING prerequisite: extend the lookahead canary

`tests/test_features.py` currently mutates **only** `close`, `high` and `volume` post-cutoff:

```sql
UPDATE prices SET close = close * 3, high = high * 3, volume = volume * 7 WHERE date > ?
```

`range_20d` reads `low`, and `gap_prior` reads `open`. **A leak through either column would pass the canary vacuously today.** The canary must mutate `open` and `low` in the SAME PR as these features — cheap here (same table, same test), unlike the intraday case which needs a whole new mutation surface.

Do not route around it, and do not cite existing canary coverage for columns it does not touch.

## Other requirements

- Added to `FEATURE_COLUMNS`; label/metadata columns unchanged.
- Every column must vary on the seeded conftest universe — sklearn's binner crashes on constant or all-NaN columns (CLAUDE.md).
- `dist_52w_high` needs 252 bars; symbols with less history get NULL, which the existing dropped-column semantics handle. Confirm it does not become all-NaN on the test fixture.
- `nullif` guards on every divisor (`prev_close`, `open`, the volume means) — a zero denominator must yield NULL, not inf.
- Tests land in the same PR, including the canary extension.
- Then `twopercent benchmark` so the referee scores the new feature set against the recorded champion. **The referee decides**, not the author.

## What this tests

If AUC and lift do not move with six new price-derived features, that is strong evidence the ceiling is real and the remaining signal lives outside daily OHLCV — which would justify the cost of #79 phase 2 (prior-day intraday features, needing its own canary extension) or a new data source. Either way the result is informative.

Note the promotion band now scales with the recorded family (#60), so adding configs raises the bar automatically rather than quietly lowering it.
